--- a/svm-toy/qt/Makefile
+++ b/svm-toy/qt/Makefile
@@ -1,19 +1,25 @@
 # use  ``export QT_SELECT=qt5'' in a command window for using qt5
 # may need to adjust the path of header files
-CXX? = g++
-INCLUDE = /usr/include/x86_64-linux-gnu/qt5
-CFLAGS = -Wall -O3 -I$(INCLUDE) -I$(INCLUDE)/QtWidgets -I$(INCLUDE)/QtGui -I$(INCLUDE)/QtCore -fPIC -std=c++11
+SHVER?=2
+CFLAGS = -Wall -O3 `pkg-config --cflags --libs Qt5Widgets`
+MOC=${MOC_PATH}
+LOCAL_LIBDIR=../..
+LIBS = -L${LOCAL_LIBDIR} -lsvm
+LIBSVM_SO_SHVER=${LOCAL_LIBDIR}/libsvm.so.$(SHVER)
 LIB = -lQt5Widgets -lQt5Gui -lQt5Core
 MOC = /usr/bin/moc
 
-svm-toy: svm-toy.cpp svm-toy.moc ../../svm.o
-	$(CXX) $(CFLAGS) svm-toy.cpp ../../svm.o -o svm-toy $(LIB)
+all: svm-toy-qt
+
+svm-toy-qt: svm-toy.cpp svm-toy.moc $(LIBSVM_SO_SHVER)
+	$(CXX) $(CFLAGS) svm-toy.cpp ${LIBS} -o $@
 
 svm-toy.moc: svm-toy.cpp
 	$(MOC) svm-toy.cpp -o svm-toy.moc
 
-../../svm.o: ../../svm.cpp ../../svm.h
-	make -C ../.. svm.o
+$(LIBSVM_SO_SHVER): ${LOCAL_LIBDIR}/svm.cpp ${LOCAL_LIBDIR}/svm.h
+	make -C ${LOCAL_LIBDIR} lib
 
 clean:
-	rm -f *~ svm-toy svm-toy.moc ../../svm.o
+	rm -f *~ svm-toy svm-toy.moc $(LIBSVM_SO_SHVER)
+
