--- a/java/Makefile
+++ b/java/Makefile
@@ -4,14 +4,17 @@
 		libsvm/svm_print_interface.class \
 		svm_train.class svm_predict.class svm_toy.class svm_scale.class
 
+JAVA_DOC_DIR=${INSTDIR}/share/javadoc/libsvm-${LIBSVM_VER}
+CLASSPATH = .
 #JAVAC = jikes
-JAVAC_FLAGS = --release 11
-JAVAC = javac
+JAVAC_FLAGS =  --release 11  -classpath ${CLASSPATH}
+JAVAC ?= javac
 # JAVAC_FLAGS =
+JAR ?= jar
 export CLASSPATH := .:$(CLASSPATH)
 
 all: $(FILES)
-	jar cvf libsvm.jar *.class libsvm/*.class
+	${JAR} cvf libsvm.jar *.class libsvm/*.class
 
 .java.class:
 	$(JAVAC) $(JAVAC_FLAGS) $<
@@ -19,8 +22,21 @@
 libsvm/svm.java: libsvm/svm.m4
 	m4 libsvm/svm.m4 > libsvm/svm.java
 
+javadoc: docs/index.html
+
+docs/index.html: $(FILES)
+	javadoc -d docs *.java libsvm/*.java
+
+install:
+	mkdir -p ${JAVA_TARGET_DIR}
+	install -m 644 libsvm.jar ${JAVA_TARGET_DIR}
+
+uninstall:
+	rm -fr ${JAVA_TARGETDIR}
+
 clean:
 	rm -f libsvm/*.class *.class *.jar libsvm/*~ *~ libsvm/svm.java
+	rm -rf docs
 
 dist: clean all
 	rm *.class libsvm/*.class

