diff -up ./svm-toy/gtk/Makefile.svm-toy ./svm-toy/gtk/Makefile
--- ./svm-toy/gtk/Makefile.svm-toy	2011-05-11 12:15:45.000000000 +1000
+++ ./svm-toy/gtk/Makefile	2012-04-04 17:02:28.366615284 +1000
@@ -1,10 +1,12 @@
 CC? = gcc
 CXX? = g++
-CFLAGS = -Wall -O3 -g `pkg-config --cflags gtk+-2.0`
-LIBS = `pkg-config --libs gtk+-2.0`
+CFLAGS = -Wall -O3 -g `pkg-config --cflags --libs gtk+-2.0`
+LOCAL_LIBDIR=../..
+LIBS = -L${LOCAL_LIBDIR} -lsvm
 
-svm-toy: main.o interface.o callbacks.o ../../svm.o
-	$(CXX) $(CFLAGS) main.o interface.o callbacks.o ../../svm.o -o svm-toy $(LIBS)
+
+svm-toy-gtk: main.o interface.o callbacks.o ${LOCAL_LIBDIR}/svm.o
+	$(CXX) $(CFLAGS) main.o interface.o callbacks.o $(LIBS) -o $@
 
 main.o: main.c
 	$(CC) $(CFLAGS) -c main.c
@@ -15,8 +17,8 @@ interface.o: interface.c interface.h
 callbacks.o: callbacks.cpp callbacks.h
 	$(CXX) $(CFLAGS) -c callbacks.cpp
 
-../../svm.o: ../../svm.cpp ../../svm.h
-	make -C ../.. svm.o
+${LOCAL_LIBDIR}/svm.o: ${LOCAL_LIBDIR}/svm.cpp ${LOCAL_LIBDIR}/svm.h
+	make -C ${LOCAL_LIBDIR} svm.o
 
 clean:
-	rm -f *~ callbacks.o svm-toy main.o interface.o callbacks.o ../../svm.o
+	rm -f *~ callbacks.o svm-toy main.o interface.o callbacks.o ${LOCAL_LIBDIR}/svm.o
diff -up ./svm-toy/qt/Makefile.svm-toy ./svm-toy/qt/Makefile
--- ./svm-toy/qt/Makefile.svm-toy	2011-05-11 12:15:45.000000000 +1000
+++ ./svm-toy/qt/Makefile	2012-04-04 17:02:28.366615284 +1000
@@ -1,17 +1,22 @@
 CXX? = g++
-CFLAGS = -Wall -O3 -I$(INCLUDE) -I$(INCLUDE)/QtGui -lQtGui
-INCLUDE = /usr/include/qt4
-MOC = /usr/bin/moc-qt4
+SHVER?=2
+CFLAGS = -Wall -O3 `pkg-config --cflags --libs QtGui`
+MOC=${MOC_PATH}
+LOCAL_LIBDIR=../..
+LIBS = -L${LOCAL_LIBDIR} -lsvm
+LIBSVM_SO_SHVER=${LOCAL_LIBDIR}/libsvm.so.$(SHVER)
 
-svm-toy: svm-toy.cpp svm-toy.moc ../../svm.o
-	$(CXX) $(CFLAGS) svm-toy.cpp ../../svm.o -o svm-toy
+all: svm-toy-qt
+
+svm-toy-qt: svm-toy.cpp svm-toy.moc $(LIBSVM_SO_SHVER)
+	$(CXX) $(CFLAGS) svm-toy.cpp ${LIBS} -o $@
 
 svm-toy.moc: svm-toy.cpp
 	$(MOC) svm-toy.cpp -o svm-toy.moc
 
-../../svm.o: ../../svm.cpp ../../svm.h
-	make -C ../.. svm.o
+$(LIBSVM_SO_SHVER): ${LOCAL_LIBDIR}/svm.cpp ${LOCAL_LIBDIR}/svm.h
+	make -C ${LOCAL_LIBDIR} lib
 
 clean:
-	rm -f *~ svm-toy svm-toy.moc ../../svm.o
+	rm -f *~ svm-toy svm-toy.moc $(LIBSVM_SO_SHVER)
 
