--- a/python/Makefile
+++ b/python/Makefile
@@ -1,4 +1,18 @@
-all = lib
+PYTHONDIR_NAME ?= python${PYTHON_VERSION}/site-packages/libsvm
+PYTHONDIR ?= ${LIBDIR}/${PYTHONDIR_NAME}
+PYTHONDIR_INSTALL ?= ${DESTDIR}/${PYTHONDIR}
+SVM_TOOLS:= $(notdir $(wildcard ../tools/*.py))
 
-lib:
-	make -C .. lib
+all: lib
+
+lib: ../svm.cpp ../svm.h
+	make -C .. lib
+
+install: all
+	mkdir -p ${PYTHONDIR_INSTALL}
+	install -m 755 *.py ${PYTHONDIR_INSTALL}
+	install -m 755 ../tools/*.py ${PYTHONDIR_INSTALL}
+	$(foreach toolPy,$(SVM_TOOLS),ln -s ${PYTHONDIR}/$(toolPy) ${BINDIR_INSTALL}/svm-$(toolPy);)
+
+uninstall:
+	rm -fr ${LIBDIR}/${PYTHON_PATH} $(addprefix ${BINDIR_INSTALL}/svm-,$(SVM_TOOLS))
